aries_vcx/src/handlers/issuance/credential_def/mod.rs

```diff
@@ -223,29 +180,32 @@ impl CredentialDef {
                 cred_def_id,
                 cred_def_json,
                 issuer_did,
-                cred_def_payment_txn: None,
                 rev_reg,
                 state: PublicEntityStateType::Built,
             }
         )
     }
 
-    pub fn publish(self) -> VcxResult<Self> {
+    pub fn publish(mut self, tails_url: Option<&str>) -> VcxResult<Self> {
```

Comment:
I suggest we fail if `tails_url` is `None` and `self.rev_reg` is `Some`

Reply:
Good call